pkg/agent/multicast/vif_allocator.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multicast
+
+import (
+	"container/list"
+	"fmt"
+	"net"
+	"sync"
+
+	"github.com/vishvananda/netlink"
+	"k8s.io/klog/v2"
+)
+
+type vifAllocator struct {
```

Comment:
Add comments.

Reply:
Is VIF a term in the Linux IGMP or mroute implementation? Probably you should add comments to explain what is a VIF too.

And probably you should use VIF rather than Vif in names.

Reply:
comment added.

VIF stands for virtual interface in the context of multicast https://github.com/torvalds/linux/blob/d58071a8a76d779eedab38033ae4c821c30295a5/include/uapi/linux/mroute.h#L21 . The interface should be added as VIF first before configuring multicast routing related to this interface